src/ResourceManagement/ContainerRegistry/RegistryImpl.cs

```diff
@@ -6,215 +6,323 @@ namespace Microsoft.Azure.Management.ContainerRegistry.Fluent
+            return Extensions.Synchronize(() => this.RegenerateCredentialAsync(accessKeyType));
+        }
+
+        ///GENMHASH:2174CAA91FF57DF106B853211C422772:2D6D6BBDF119ED38C01A18861C7D5699
+        public Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.IRegistryCredentials> RegenerateCredentialAsync(AccessKeyType accessKeyType, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.Manager.ContainerRegistries.RegenerateCredentialAsync(this.ResourceGroupName, this.Name, accessKeyType, cancellationToken);
+        }
+
+        ///GENMHASH:1FF09F84F47B1424B14D7C169D1A59C6:3ABD076ED3162E6FAE08B80AC4D00F46
+        public IReadOnlyCollection<Models.RegistryUsage> ListQuotaUsages()
+        {
+            return Extensions.Synchronize(() => this.ListQuotaUsagesAsync());
         }
 
-        public async Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.Models.RegistryListCredentials> RegenerateCredentialAsync(PasswordName passwordName, CancellationToken cancellationToken = default(CancellationToken))
+        ///GENMHASH:5F9E03CF4D4E27AAF0427B1377B983E9:C7773E788A550788FE4CD76DC857A54F
+        public Task<IReadOnlyCollection<Models.RegistryUsage>> ListQuotaUsagesAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            return await this.Manager.Inner.Registries.RegenerateCredentialAsync(this.ResourceGroupName, this.Name, passwordName);
+            return this.Manager.ContainerRegistries.ListQuotaUsagesAsync(this.ResourceGroupName, this.Name, cancellationToken);
         }
 
+
+        ///GENMHASH:5AD91481A0966B059A478CD4E9DD9466:1F3B7DE9E8C64BB113C876629C3657C2
+        protected override Task<Models.RegistryInner> GetInnerAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.Manager.Inner.Registries.GetAsync(this.ResourceGroupName, this.Name, cancellationToken);
+        }
+
+        ///GENMHASH:0202A00A1DCF248D2647DBDBEF2CA865:82803CC7B92AB9DAB1E6ADA17D4D4332
         public override async Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.IRegistry> CreateResourceAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            RegistryImpl self = this;
             if (IsInCreateMode)
             {
-                var storageAccountParameters = await this.HandleStorageSettingsAsync(cancellationToken);
-                createParameters.StorageAccount = storageAccountParameters;
-                createParameters.Location = this.RegionName.ToLower();
-                createParameters.Tags = this.Inner.Tags;
-                var registryInner = await this.Manager.Inner.Registries.CreateAsync(this.ResourceGroupName, this.Name, createParameters);
+                if (this.creatableStorageAccountKey != null)
+                {
+                    IStorageAccount storageAccount = this.CreatedResource(this.creatableStorageAccountKey) as IStorageAccount;
```

Comment:
>as IStorageAccount [](start = 107, length = 18)

Please use direct cast.

Reply:
reworked the code so it does not need a cast.